tests/testsemaphore.nim

```diff
@@ -1,6 +1,7 @@
 import random
 import chronos
-import ../libp2p/utils/semaphore
+
+include ../libp2p/utils/semaphore # include to avoid exposing private members
```

Comment:
can import now

Reply:
I'm still using `queue` which is private

Reply:
...which is dubious - that means that the testing of the API is incomplete or taking shortcuts - testing the private implementation details means that it's not possible to use unit tests to detect backwards compatibility issues - this is a useful point to consider before 1.0: how to know if something is backwards compatible - if you need to change a test, it's _not_ backwards compatible for sure, but this litmus test depends on the tests not using hacks like this.

Reply:
I usually avoid private api tests, but in some narrow cases it makes sense. In this case it isn't 100% indispensable anymore and was used initially as a debugging/sanity mechanism which can be removed now.

As for versioning, I want to again stress - IMO, versions don't communicate stability, they communicate changes, associating software stability with a version number doesn't scale to automated package management for example.

Reply:
> stability

tests don't communicate "stability": they indicate the lack of instability in a subset of the application - but that aside, they also work as backwards incompatibility detectors: if you need to change a test that tests public api only, you're not backwards compatible and therefore need to bump the major version - it's a tool to help judge the impact of a change.

Reply:
I meant it as a general remark about versions, I don't disagree with the fact that tests communicate backwards compatibility.